File: app/src/main/java/network/devandroid/com/networklibrarycomparison/fastnetworking/FastNetworkManager.java
package network.devandroid.com.networklibrarycomparison.fastnetworking;

import android.os.Handler;
import android.os.Looper;

import com.androidnetworking.AndroidNetworking;
import com.androidnetworking.error.ANError;
import com.androidnetworking.interfaces.StringRequestListener;

import network.devandroid.com.networklibrarycomparison.internal.BaseNetworkManager;

public class FastNetworkManager extends BaseNetworkManager<String> {

    public FastNetworkManager(Callback callback) {
        super(callback);
    }

    @Override
    public void send(String url) {
        AndroidNetworking.get(url)
                .setTag("test")
                .doNotCacheResponse()
                .build()
                .getAsString(new StringRequestListener() {
                    @Override
                    public void onResponse(final String response) {
                        if (null != mCallback) {
                            new Handler(Looper.getMainLooper()).post(new Runnable() {
                                @Override
                                public void run() {
                                    mCallback.onResponse(response);
                                }
                            });
                        }
                    }

                    @Override
                    public void onError(final ANError anError) {
                        if (null != mCallback) {
                            new Handler(Looper.getMainLooper()).post(new Runnable() {
                                @Override
                                public void run() {
                                    mCallback.onError(anError.getLocalizedMessage());
                                }
                            });
                        }
                    }
                });
    }
}

File: app/src/main/java/network/devandroid/com/networklibrarycomparison/fastnetworking/FastNetworkManager.kt
package network.devandroid.com.networklibrarycomparison.fastnetworking

import android.os.Handler
import android.os.Looper

import com.androidnetworking.AndroidNetworking
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.StringRequestListener

import network.devandroid.com.networklibrarycomparison.internal.BaseNetworkManager
import network.devandroid.com.networklibrarycomparison.internal.INetworkManager

class FastNetworkManager(callback: INetworkManager.Callback) : BaseNetworkManager(callback) {

    override fun send(url: String) {
        AndroidNetworking.get(url)
                .setTag("test")
                .doNotCacheResponse()
                .build()
                .getAsString(object : StringRequestListener {
                    override fun onResponse(response: String) {
                        if (null != callback) {
                            Handler(Looper.getMainLooper()).post { callback.onResponse(response) }
                        }
                    }

                    override fun onError(anError: ANError) {
                        if (null != callback) {
                            Handler(Looper.getMainLooper()).post { callback.onError(anError.localizedMessage) }
                        }
                    }
                })
    }
}
